### ISSUE TITLE
Plugin does not work with WebStorm

### ISSUE THREAD

**Issue by jbe456** [benmosher/eslint-plugin-import#93]
Using Webstorm 11 under Windows, the plugin does not work although it works using the console.

When using webstorm integrated eslint here is the command launched:
```
"C:\Program Files\nodejs\node.exe" C:\Users\XXX\workspace\myproject\node_modules\eslint\bin\eslint.js --format=checkstyle --config C:\Users\XXX\workspace\myproject\.eslintrc --stdin --stdin-filename path\to\file.js
```

The file to inspect is thus given as a relative path. Then in `resolve.js`, while using `resolver.sync` method, all paths are resolved with a leading `\`.

For example, in method `loadNodeModulesSync` of file `sync.js` `resolve` dependency: 
```
dirs=["/C:\\Users\\XXX\\...", ...];
```
Then the method `loadAsFileSync` is called with:
`path.join(dir, '/', x) = \C:\Users\XXX\...`

Then the `isFile` method can't resolve the file, because the leading `\` gives the following internal catched error:
```
ENOENT: no such file or directory, stat 'C:\C:\Users\XXX\...'
```

Not sure if it is a bug of resolve or if their api is misuse. Any idea?

PS: meanwhile I workaround the issue by passing a custom `isFile` method and modifying the core resolve code:
```
function opts(basedir, settings) {
  // pulls all items from 'import/resolve'
  return _Object$assign(
    {},
    settings['import/resolve'],
    {
      basedir: basedir,
      isFile: function(file) {
        let stat;

        // remove leading backslash if any ...
        if (file.startsWith('\\')) {
          file = file.substring(1);
        }

        try {
          stat = fs.statSync(file);
        } catch (err) {
          if (err && err.code === 'ENOENT') { return false; }
        }
        return stat.isFile() || stat.isFIFO();
      }
    }
  );
}
```

and in `core\resolve.js`:
```
try {
      var file = resolver.sync(p, opts(path.dirname(context.getFilename()), context.settings));

      // remove leading backslash if any ...
      if (file.startsWith('\\')) {
        file = file.substring(1);
      }

      if (!fileExists(file)) return null;
      return file;
    }
```
This is obviously a hack waiting for a long term solution

**Comment by benmosher** [comment]
Interesting. I've seen some weirdness with PyCharm on OS X as well. I think the ESLint integration guts are the same.

Thanks for all the forensics. 😎 Not sure how to move on this but I'll dig in.

**Comment by benmosher** [comment]
I haven't had a chance to look at this--been going all in on the Webpack resolver.

I have a hope--probably an unreasonable hope--that the Webpack resolver might solve some of this. Depending on your config, I think it might help.

**Comment by benmosher** [comment]
BTW: I tried to open up PyCharm on OS X to play around and see if I could see similar behavior... unfortunately, it just ate 53GB (GB!) of memory and crashed my machine ("you have run out of working memory") after cooking my legs at maximum fan speed for 10 minutes.

Reminds me of why I switched to Sublime. 😅

Let me know if the new Webpack resolver helps. If not, I'll poke around more in your fork and see what I can intuit.
